zvariant_utils/src/case.rs

```diff
@@ -26,16 +26,60 @@ pub fn pascal_or_camel_case(s: &str, is_pascal_case: bool) -> String {
     result
 }
 
-/// Convert to snake case, assuming pascal case.
```

Comment:
You're passing `true` to `pascal_or_camel_case`, which tells it that passed string is pascal case, so isn't pascal case still assumed?

Reply:
Oh, I got confused because of multiple case changes. Still, this function now assume pascal or came case now, doesn't it?

Reply:
No not really anymore, it actually just converts the input to PascalCase  using `pascal_or_camel_case(s, true)`. So it doesn't matter what the input is anymore.

Reply:
So I can pass it a string in any format and it can handle it? `pascal_or_camel_case` itself can't handle all cases so I'm even more confused now. :thinking:

Reply:
I update the comments. But basicaly yes all the functions can handle all forms of valid casing. See the test cases.